### PR TITLE
Allow viewing webms via webm streaming in local threads

### PR DESCRIPTION
**Pull request by K1rakishou**

### CODE REVIEW

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/stream/WebmStreamingSource.kt [hunk 1]
```diff
@@ -110,6 +96,164 @@ class WebmStreamingSource(
...
+
+        BackgroundUtils.runOnMainThread {
+            callback.onMediaSourceReady(
+                    ProgressiveMediaSource.Factory(DataSource.Factory { FileDataSource() })
+                            .createMediaSource(fileUri)
+            )
+        }
+    }
+
+    private fun createDestinationFile(
```
Comment (Adamantcheese) [inline Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/stream/WebmStreamingSource.kt]: This function is remarkably similar to FileCacheV2's handleLocalThreadFileImmediatelyAvailable, maybe the two can be combined into one?

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/stream/WebmStreamingSource.kt [hunk 2]
```diff
@@ -110,6 +96,164 @@ class WebmStreamingSource(
...
+        Logger.d(TAG, "createMediaSource() Loading already downloaded file from the disk")
+        val fileUri = Uri.parse(rawFile.getFullPath())
+
+        callback.onMediaSourceReady(
+                ProgressiveMediaSource.Factory(DataSource.Factory { FileDataSource() })
+                        .createMediaSource(fileUri)
+        )
+    }
+
+    private fun loadFromLocalThread(
```
Comment (Adamantcheese) [inline Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/stream/WebmStreamingSource.kt]: This seems remarkably similar to FileCacheV2's handleLocalThreadFile, maybe the two can be combined into one?

**Review by Adamantcheese — CHANGES_REQUESTED** [review]

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/stream/WebmStreamingSource.kt (hunk 1 above)
Comment (K1rakishou) [reply Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/stream/WebmStreamingSource.kt]: Yeah, they are pretty much the same. Good catch!

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/stream/WebmStreamingSource.kt (hunk 2 above)
Comment (K1rakishou) [reply Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/stream/WebmStreamingSource.kt]: Same.

**Pull request merged**